edgraph/server.go

```diff
@@ -1883,6 +1884,356 @@ func (s *Server) CheckVersion(ctx context.Context, c *api.Check) (v *api.Version
+		return nil, err
+	}
+	dataNsList := make(map[uint64]string)
+	for _, e := range data.NsList {
+		dataNsList[e.ID] = e.Name
+	}
+
+	schNsList := schema.State().Namespaces()
+	result := &apiv25.ListNamespacesResponse{NsList: make(map[string]*apiv25.Namespace)}
+	for id := range schNsList {
+		if name, ok := dataNsList[id]; !ok {
+			name = fmt.Sprintf("dgraph-%d", id)
+			result.NsList[name] = &apiv25.Namespace{Name: name, Id: id}
+		} else {
+			result.NsList[name] = &apiv25.Namespace{Name: name, Id: id}
+		}
+	}
+
+	return result, nil
+}
+
+// verifyNamespaceName ensures that name only has alpha numeric characters as
+// well as underscores and hyphens. It also ensures that the name is not empty.
+func verifyNamespaceName(name string) error {
+	if name == "" {
+		return errors.Errorf("namespace name cannot be empty")
+	}
+	hasInvalidChars := strings.ContainsFunc(name, func(r rune) bool {
+		return !unicode.IsLetter(r) && !unicode.IsNumber(r) && r != '_' && r != '-'
+	})
+	if hasInvalidChars {
+		return fmt.Errorf("namespace name [%v] has invalid characters", name)
+	}
+	if strings.HasPrefix(name, "_") || strings.HasPrefix(name, "-") {
+		return fmt.Errorf("namespace name [%v] cannot start with _ or -", name)
+	}
+	if strings.HasPrefix(name, "dgraph") {
+		return fmt.Errorf("namespace name [%v] cannot start with dgraph", name)
+	}
+	if name == "system" || name == "galaxy" {
+		return fmt.Errorf("namespace name [%v] is reserved", name)
+	}
+	if _, err := strconv.ParseInt(name, 10, 64); err == nil {
+		return fmt.Errorf("namespace name [%v] cannot be a number", name)
+	}
+	return nil
+}
+
+func verifyDropNamespaceName(name string) error {
+	if name == "" {
+		return errors.Errorf("namespace name cannot be empty")
+	}
+	if name == "system" || name == "galaxy" || name == "dgraph-0" {
+		return fmt.Errorf("namespace [%v] cannot be renamed/dropped", name)
+	}
+	return nil
+}
+
+func getNamespaceID(ctx context.Context, nsName string) (uint64, error) {
+	const q = `{q(func: eq(dgraph.namespace.name, "%v")) { dgraph.namespace.id }}`
+	req := &api.Request{Query: fmt.Sprintf(q, nsName)}
+	resp, err := (&Server{}).doQuery(ctx, &Request{req: req, doAuth: NoAuthorize})
+	if err != nil {
+		return 0, err
+	}
+
+	var data struct {
+		Data []struct {
+			ID uint64 `json:"dgraph.namespace.id"`
+		} `json:"q"`
+	}
+	if err := json.Unmarshal(resp.GetJson(), &data); err != nil {
+		return 0, err
+	}
+	if len(data.Data) == 0 {
+		return 0, errors.Errorf("namespace %q not found", nsName)
+	}
+
+	glog.Infof("Found namespace [%v] with id [%d]", nsName, data.Data[0].ID)
+	return data.Data[0].ID, nil
+}
+
+func getNamespaceName(ctx context.Context, nsID uint64) (string, error) {
+	const q = `{q(func: eq(dgraph.namespace.id, "%v")) { dgraph.namespace.name }}`
+	req := &api.Request{Query: fmt.Sprintf(q, nsID)}
+	resp, err := (&Server{}).doQuery(ctx, &Request{req: req, doAuth: NoAuthorize})
+	if err != nil {
+		return "", err
+	}
+
+	var data struct {
+		Data []struct {
```

Comment:
Make it one function instead of two. get booth name and id at the same time. Then you can get name or id whatever you want out of it